src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
Well, `CacheLoader` is an abstract base class. I would say that anyone extending it should narrow the types of checked extensions which could possibly be thrown. It still sucks for consumers though when all they know is they have a `CacheLoader` of some type, and now they need to `catch (Exception e)` in order for their code to compile.

Exception handling is a bit of a "religious" issue I think, and maybe there are counterarguments. I dunno.